### ISSUE TITLE
Segfault on fetch_timezone_offset with ApacheBench

### ISSUE THREAD

**Issue by nikkiii** [facebook/hhvm#6565]
While testing my new application using Laravel 5.1 and ApacheBench, HHVM Segfaults on fetch_timezone_offset (as a result of exception logging) when requested through ApacheBench. This is due to exception logging somewhere in Laravel, which ends up trying to create a DateTime object from a timestamp.

It happens in hhvm 3.10.1 (stable), 3.11.0-dev (11-17-2015 nightly)

Versions:
`ApacheBench, Version 2.3 <$Revision: 1528965 $>`

Setup:
Ubuntu 14.04.3 LTS (LXC), Nginx 1.9.6

Backtrace:
```
#0  0x000000000589f02f in fetch_timezone_offset (tz=0x0, ts=1447811025, transition_time=0x7fffcb7f9138) at /tmp/tmp.ICxnvRXYXb/third-party/timelib/parse_tz.c:861
#1  0x000000000589f2ee in timelib_get_time_zone_info (ts=1447811025, tz=0x0) at /tmp/tmp.ICxnvRXYXb/third-party/timelib/parse_tz.c:935
#2  0x00000000058a212d in timelib_unixtime2local (tm=0x7fffcba6a300, ts=1447811025) at /tmp/tmp.ICxnvRXYXb/third-party/timelib/unixtime2tm.c:194
#3  0x0000000005c5ef76 in HPHP::DateTime::fromTimeStamp (this=0x7fffc64914e0, timestamp=1447811025, utc=false) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/datetime.cpp:304
#4  0x0000000003a6fdeb in HPHP::DateTime::DateTime (this=0x7fffc64914e0, timestamp=1447811025, tz=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/datetime.cpp:277
#5  0x0000000004f838ac in HPHP::req::make<HPHP::DateTime, long, HPHP::req::ptr<HPHP::TimeZone> >(long&&, HPHP::req::ptr<HPHP::TimeZone>&&) () at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/resource-data.h:303
#6  0x0000000004f7f45d in HPHP::c_DateTime_ni___construct (this_=0x7fffc6421840, time=..., timezone=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/ext/datetime/ext_datetime.cpp:105
#7  0x0000000005c7ad4a in HPHP::Native::callFuncInt64Impl (f=0x4f7f3bb <HPHP::c_DateTime_ni___construct(HPHP::ObjectData*, HPHP::String const&, HPHP::Variant const&)>, GP=0x7fffcb7f95d0, GP_count=3, SIMD=0x7fffcb7f9590, SIMD_count=0)
    at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/native-func-caller.h:810
#8  0x00000000040c864b in HPHP::Native::callFunc<false> (func=0x7fffe75f9c00, ctx=0x7fffc6421840, args=0x7fffbd6bfba0, numNonDefault=1, ret=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/native.cpp:222
#9  0x00000000040c6bf3 in HPHP::Native::methodWrapper<false> (ar=0x7fffbd6bfbb0) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/native.cpp:444
#10 0x0000000003edb00f in HPHP::iopNativeImpl (pc=@0x7fffcb7f9920: 0x7fffe52310bb "\352\030") at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:7285
#11 0x0000000003edf9b8 in HPHP::iopRetWrapper (fn=0x3edaf98 <HPHP::iopNativeImpl(unsigned char const*&)>, pc=@0x7fffcb7f9920: 0x7fffe52310bb "\352\030") at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:7876
#12 0x0000000005c181c9 in HPHP::dispatchImpl<false> () at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:8052
#13 0x0000000003f34237 in HPHP::dispatch () at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:8069
#14 0x0000000003eb583c in HPHP::enterVMAtFunc (enterFnAr=0x7fffbd6bffc0, stk=HPHP::Untrimmed, varEnv=0x7fffc6407320) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2066
#15 0x0000000003eb5a37 in HPHP::ExecutionContext::__lambda136::operator() (__closure=0x7fffcb7f9a60) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2236
#16 0x0000000003f7aafe in HPHP::exception_handler<HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda136>(HPHP::ExecutionContext::__lambda136) (action=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/unwind-inl.h:30
#17 0x0000000003f35791 in HPHP::__lambda134::operator() (__closure=0x7fffcb7f9b10) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2110
#18 0x0000000003f7b22c in HPHP::enterVMCustomHandler<HPHP::enterVM(HPHP::ActRec*, Action) [with Action = HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda136]::__lambda134>(HPHP::ActRec *, HPHP::__lambda134) (ar=0x7fffbd6bffc0, action=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2101
#19 0x0000000003f357c6 in HPHP::enterVM<HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, const HPHP::Func*, const HPHP::Variant&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags)::__lambda136>(HPHP::ActRec *, HPHP::ExecutionContext::__lambda136) (ar=0x7fffbd6bffc0, action=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2110
#20 0x0000000005c34870 in HPHP::ExecutionContext::invokeFunc (this=0x7fffc6406010, retptr=0x7fffcb7f9de0, f=0x7fffe08420c0, args_=..., this_=0x0, cls=0x0, varEnv=0x7fffc6407320, invName=0x0, flags=HPHP::ExecutionContext::InvokePseudoMain)
    at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2237
#21 0x0000000005c6b3ce in HPHP::ExecutionContext::invokeUnit (this=0x7fffc6406010, retval=0x7fffcb7f9de0, unit=0x7fffe77ea100) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/vm/bytecode.cpp:2407
#22 0x0000000003b2773b in HPHP::invoke_file_impl (res=..., path=..., once=true, currentDir=0x5d6915c "") at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/builtin-functions.cpp:806
#23 0x0000000005c6b2e9 in HPHP::invoke_file (s=..., once=true, currentDir=0x5d6915c "") at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/builtin-functions.cpp:819
#24 0x0000000003b2783a in HPHP::include_impl_invoke (file=..., once=true, currentDir=0x5d6915c "") at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/builtin-functions.cpp:830
#25 0x0000000003c819b8 in HPHP::hphp_invoke (context=0x7fffc6406010, cmd=..., func=false, funcParams=..., funcRet=..., reqInitFunc=..., reqInitDoc=..., error=@0x7fffcb7fa059: false, errorMsg=..., once=true, warmupOnly=false, richErrorMsg=false)
    at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/program-functions.cpp:2112
#26 0x0000000003d6e0f2 in HPHP::HttpRequestHandler::executePHPRequest (this=0x7fffcbb4bb50, transport=0x7fffe87e2b98, reqURI=..., sourceRootInfo=..., cacheableDynamicContent=false) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/server/http-request-handler.cpp:477
#27 0x0000000003d6d696 in HPHP::HttpRequestHandler::handleRequest (this=0x7fffcbb4bb50, transport=0x7fffe87e2b98) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/server/http-request-handler.cpp:388
#28 0x0000000003dd59d3 in HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJobImpl (this=0x7fffe0f88a80, job=..., abort=false) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/server/server-worker.h:108
#29 0x0000000003dd520e in HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJob (this=0x7fffe0f88a80, job=...) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/server/server-worker.h:56
#30 0x0000000003dd1c77 in HPHP::JobQueueWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::Server*, true, false, HPHP::JobQueueDropVMStack>::start (this=0x7fffe0f88a80) at /tmp/tmp.ICxnvRXYXb/hphp/util/job-queue.h:463
#31 0x0000000003cc450e in HPHP::AsyncFunc<HPHP::LibEventHttpClient>::run_ (obj=0x7fffe0fb5000) at /tmp/tmp.ICxnvRXYXb/hphp/util/async-func.h:213
#32 0x000000000543cc99 in HPHP::AsyncFuncImpl::threadFuncImpl (this=0x7fffe0fb5000) at /tmp/tmp.ICxnvRXYXb/hphp/util/async-func.cpp:131
#33 0x000000000543c88b in HPHP::AsyncFuncImpl::ThreadFunc (obj=0x7fffe0fb5000) at /tmp/tmp.ICxnvRXYXb/hphp/util/async-func.cpp:51
#34 0x0000000003bc5d69 in HPHP::start_routine_wrapper (arg=0x7fffe0fc4400) at /tmp/tmp.ICxnvRXYXb/hphp/runtime/base/thread-hooks.cpp:104
#35 0x00007ffff1332182 in start_thread (arg=0x7fffcb7ff700) at pthread_create.c:312
#36 0x00007ffff083f47d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

**Comment by jwatzman** [comment]
I wonder if this is the same root cause as #6594 -- that has a different stack trace, but in a related area, and if the issue is "missing a mutex" then I could def see it being related.

**Comment by aorenste** [comment]
(Hopefully) fixed in #6594
